pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -200,17 +209,20 @@ type stringSliceFuncJob struct {
 	warnings storage.Warnings
 }
 
-// mergeDistinctStringSlice is aggregating results from stringSliceFunc calls
-// on per querier in parallel. It removes duplicates and sorts the result. It
-// doesn't require the output of the stringSliceFunc to be sorted, as results
+// mergeDistinctStringSliceWithLabelValues aggregates stringSlinceFunc call
+// results from queriers whose ids match the labelValues. If a nil map is
+// provided, all queriers are used. It removes duplicates and sorts the result.
+// It doesn't require the output of the stringSliceFunc to be sorted, as results
 // of LabelValues are not sorted.
-func (m *mergeQuerier) mergeDistinctStringSlice(f stringSliceFunc) ([]string, storage.Warnings, error) {
-	var jobs = make([]interface{}, len(m.ids))
+func (m *mergeQuerier) mergeDistinctStringSliceWithLabelValues(f stringSliceFunc, labelValues map[string]struct{}) ([]string, storage.Warnings, error) {
```

Comment:
"withLabelValues" seems very generic; as I understand it the code is specifically matching tenant IDs

Reply:
I agree, I have changed this to `mergeDistinctStringSliceWithTenants` in 0dc3ef4
along with some other changes that makes it clearer that we are passing around a map of matched tenant IDs